test/bot.test.js

```diff
@@ -39,6 +39,14 @@ describe('Bot', function() {
     sandbox.restore();
   });
 
+  const createServerStub = function(nickname) {
+    return {
+      detailsOfUser(username) {
+        return { nick: nickname };
+      }
+    };
+  };
+
```

Comment:
Maybe I wrote something wrong in the last one, here's how it would be with an arrow function at least:

```js
const createServerStub = (nickname = null) => ({
  detailsOfUser() {
    return { nick: nickname };
  }
});
```

Note the `nickname = null`, that way you wont have to explicitly call it with `null` since it defaults to it (you could do this with your code to as well, don't need an arrow function - it just looks a bit cleaner).